seqnum/seqnum_test.go

```diff
@@ -0,0 +1,113 @@
+package seqnum_test
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/Azure/custom-script-extension-linux/seqnum"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSet_nonExistingDir(t *testing.T) {
+	err := seqnum.Set("/non/existing/path", 1)
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "seqnum: failed to write")
+}
+
+func TestSet_writeFail(t *testing.T) {
+	fp := testFile(t, 0500) // remove read permissions // remove write permissions
+	defer os.RemoveAll(fp)
+
+	err := seqnum.Set(fp, 0)
+	require.NotNil(t, err)
+	require.Contains(t, err.Error(), "seqnum: failed to write")
+}
+
+func TestSet_newFile(t *testing.T) {
+	f, err := ioutil.TempFile("", "")
+	require.Nil(t, err)
+	f.Close()
+	fp := f.Name()
+	defer os.RemoveAll(fp)
+
+	require.Nil(t, seqnum.Set(fp, 1))
+
+	// validate contents
+	b, err := ioutil.ReadFile(fp)
+	require.Nil(t, err)
+	require.Equal(t, "1", string(b))
+
+	// validate chmod
+	fi, err := os.Stat(fp)
+	require.Nil(t, err)
+	require.EqualValues(t, os.FileMode(0600).String(), fi.Mode().String())
+}
+
+func TestSet_truncates(t *testing.T) {
+	f, err := ioutil.TempFile("", "")
+	require.Nil(t, err)
+	f.Close()
+	fp := f.Name()
+	defer os.RemoveAll(fp)
+
+	require.Nil(t, seqnum.Set(fp, 1))
+	require.Nil(t, seqnum.Set(fp, 2))
+
+	b, err := ioutil.ReadFile(fp)
+	require.Nil(t, err)
+	require.Equal(t, "2", string(b))
+}
+
+func TestSetGet(t *testing.T) {
+	fp := testFile(t, 0600)
+	defer os.RemoveAll(fp)
+
+	require.Nil(t, seqnum.Set(fp, 1))
+
+	b, err := seqnum.IsSmallerThan(fp, 2)
+	require.Nil(t, err)
+	require.True(t, b, "1<2")
+
+	b, err = seqnum.IsSmallerThan(fp, 1)
+	require.Nil(t, err)
+	require.False(t, b, "1≮1")
```

Comment:
@boumenot your font might be tricking you.

![image](https://cloud.githubusercontent.com/assets/159209/17002415/04104306-4e80-11e6-98b8-58d07025c303.png)